src/main/java/org/broadinstitute/hellbender/tools/exome/SegmentUtils.java

```diff
@@ -13,292 +15,452 @@
-     * falls in the middle of a target.  Since sites in {@code snps} are length-1, there is no such concern for them.
+     * Given an interval and collections of targets and SNPs, returns a trimmed interval produced by removing the empty
+     * portions at the start and the end of the original interval that do not overlap the targets and SNPs that overlap
+     * with the original interval.  If this procedure would remove the entire interval, the original interval is
+     * returned instead.
      */
-    public static SimpleInterval parsimoniousInterval(final Locatable interval, final TargetCollection<? extends Locatable> targets,
-                                                                     final TargetCollection<? extends Locatable> snps) {
-        Utils.nonNull(interval, "Interval may not be null");
-        Utils.nonNull(targets, "Targets may not be null");
-        Utils.nonNull(snps, "SNPs may not be null");
+    public static SimpleInterval trimInterval(final Locatable interval,
+                                              final TargetCollection<? extends Locatable> targets,
+                                              final TargetCollection<? extends Locatable> snps) {
+        Utils.nonNull(interval, "The interval cannot be null.");
+        Utils.nonNull(targets, "The collection of targets cannot be null.");
+        Utils.nonNull(snps, "The collection of SNPs cannot be null.");
 
         final IndexRange targetRange = targets.indexRange(interval);
         final IndexRange snpRange = snps.indexRange(interval);
 
-        final int targetsInInterval = targetRange.size();
-        final int snpsInInterval = snpRange.size();
+        final int numTargetsInInterval = targetRange.size();
+        final int numSNPsInInterval = snpRange.size();
 
-        int start;
-        int end;
+        int start = interval.getStart();
+        int end = interval.getEnd();
 
-        if ( targetsInInterval == 0 && snpsInInterval == 0) {
-            start = interval.getStart();
-            end = interval.getStart();
-        } else if (targetsInInterval == 0) {
+        if (numTargetsInInterval == 0 && numSNPsInInterval > 0) {
+            //if there are no targets overlapping interval, use SNPs to determine trimmed interval
             start = snps.target(snpRange.from).getStart();
             end = snps.target(snpRange.to - 1).getEnd();
-        } else if (snpsInInterval == 0) {
-            start = targets.target(targetRange.from).getStart();
-            end = targets.target(targetRange.to - 1).getEnd();
-        } else {
-            start = Math.min(targets.target(targetRange.from).getStart(), snps.target(snpRange.from).getStart());
-            end = Math.max(targets.target(targetRange.to - 1).getEnd(), snps.target(snpRange.to - 1).getEnd());
+        } else if (numTargetsInInterval > 0) {
+            if (!(targets.target(targetRange.from).getStart() <= interval.getStart() &&
```

Comment:
I'm confused -- the code checks whether the interval starts before the first target or after the first target ends.  But isn't the latter impossible since the first target was derived from `targets.indexRange(interval).from`?  i.e. if a target ends before the interval starts, how can the target be part of the interval's associated index range?  So, it seems that you only need a single condition: `if (start < firstTarget.getStart())`.